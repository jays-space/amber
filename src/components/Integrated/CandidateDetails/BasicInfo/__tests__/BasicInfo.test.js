import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import { BasicInfo } from "..";

describe("<BasicInfo />", () => {
  const mockBasicInfo = {
    fullName: {
      title: "Mr",
      first: "John",
      last: "Doe",
    },
    dob: {
      date: "1964-04-24T06:25:40.777Z",
      age: 57,
    },
    gender: "male",
    email: "my.email@gmail.com",
    phone: "011 123 45 78",
    cell: "078 123 45 67",
  };

  it("should render correctly", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const basicInfoComponent = screen.getByTitle("about-candidate");
    expect(basicInfoComponent).toBeInTheDocument();
    expect(basicInfoComponent).toMatchSnapshot();
  });

  it("should render appropriate heading", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const headingElement = screen.getByRole("heading", { name: "basic info" });
    expect(headingElement).toBeInTheDocument();
  });

  it("should render full name details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const fullNameComponent = screen.getByTestId(
      "full name-details"
    );
    expect(fullNameComponent).toBeInTheDocument();
  });

  it("should render gender details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const genderComponent = screen.getByTestId(
      "gender-details"
    );
    expect(genderComponent).toBeInTheDocument();
  });

  it("should render dob details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const dobComponent = screen.getByTestId(
      "dob-details"
    );
    expect(dobComponent).toBeInTheDocument();
  });

  it("should render email details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const emailComponent = screen.getByTestId(
      "email-details"
    );
    expect(emailComponent).toBeInTheDocument();
  });

  it("should render phone details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const phoneComponent = screen.getByTestId(
      "phone-details"
    );
    expect(phoneComponent).toBeInTheDocument();
  });

  it("should render cell details", () => {
    render(<BasicInfo basicInfo={mockBasicInfo} />);

    const cellComponent = screen.getByTestId(
      "cell-details"
    );
    expect(cellComponent).toBeInTheDocument();
  });

});
