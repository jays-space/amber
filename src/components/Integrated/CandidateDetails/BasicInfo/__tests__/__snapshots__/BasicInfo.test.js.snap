// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BasicInfo /> should render correctly 1`] = `
<div
  title="about-candidate"
>
  <h4
    class="mb-6 md:mb-2 text-slate-700 capitalize text-xl font-bold"
  >
    basic info
  </h4>
  <div
    class="md:p-4 grid grid-rows-1 gap-6 md:border rounded-md"
  >
    <div
      class="flex flex-col "
      data-testid="full name-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        full name
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          John
        </p>
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          Doe
        </p>
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          (Mr)
        </p>
      </span>
    </div>
    <div
      class="flex flex-col "
      data-testid="gender-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        gender
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          male
        </p>
      </span>
    </div>
    <div
      class="flex flex-col "
      data-testid="dob-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        dob
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          Fri, 24 Apr 1964
        </p>
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          (57)
        </p>
      </span>
    </div>
    <div
      class="flex flex-col text-sm lg:text-base"
      data-testid="email-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        email
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 false"
          data-testid="paragraph"
        >
          my.email@gmail.com
        </p>
      </span>
    </div>
    <div
      class="flex flex-col "
      data-testid="phone-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        phone
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          011 123 45 78
        </p>
      </span>
    </div>
    <div
      class="flex flex-col "
      data-testid="cell-details"
    >
      <p
        class="tracking-wide  text-slate-700 mb-2 text-xs capitalize"
        data-testid="paragraph"
      >
        cell
      </p>
      <span
        class="flex"
      >
        <p
          class="tracking-wide  text-slate-700  font-semibold mr-2 last-of-type:mr-0 capitalize !text-sm !md:text-base"
          data-testid="paragraph"
        >
          078 123 45 67
        </p>
      </span>
    </div>
  </div>
</div>
`;
